tools/stats/upload_stats_lib.py

```diff
@@ -213,3 +220,130 @@ def is_rerun_disabled_tests(root: ET.ElementTree) -> bool:
 
     message = skipped.attrib.get("message", "")
     return TARGET_WORKFLOW in message or "num_red" in message
+
+
+def _convert_float_values_to_decimals(data: Dict[str, Any]) -> Dict[str, Any]:
+    for key, value in data.items():
+        if isinstance(value, float):
+            data[key] = decimal.Decimal(str(value))  # str() preserves the precision
+    return data
+
+
+class EnvVarMetric:
+    name: str
+    env_var: str
+    required: bool = True
+    # Used to cast the value of the env_var to the correct type (defaults to str)
+    type_conversion_fn: Any = None
+
+    def __init__(
+        self,
+        name: str,
+        env_var: str,
+        required: bool = True,
+        type_conversion_fn: Any = None,
+    ) -> None:
+        self.name = name
+        self.env_var = env_var
+        self.required = required
+        self.type_conversion_fn = type_conversion_fn
+
+    def value(self) -> Any:
+        value = os.environ.get(self.env_var)
+        if value is None:
+            if self.required:
+                raise ValueError(
+                    (
+                        f"Missing {self.name}. Please set the {self.env_var}"
+                        "environment variable to pass in this value."
+                    )
+                )
+            else:
+                return None
```

Comment:
Raise exception if required and value is None (also, don't you need to run it thru a conversion_fn, which might turn None into `0` or something?
```suggestion
        if value is None and self.required:
              raise ValueError(
                      f"Missing {self.name}. Please set the {self.env_var}" +
                      "environment variable to pass in this value."
              )
```